components/style/values/specified/length.rs

```diff
@@ -640,21 +640,24 @@ impl Length {
     }
 }
 
-impl<T: Parse> Either<Length, T> {
-    /// Parse a non-negative length
+/// A wrapper of Length, whose value must be >= 0.
+pub type NonNegativeLength = NonNegative<Length>;
+
+impl Parse for NonNegativeLength {
     #[inline]
-    pub fn parse_non_negative_length<'i, 't>(context: &ParserContext, input: &mut Parser<'i, 't>)
-                                             -> Result<Self, ParseError<'i>> {
-        if let Ok(v) = input.try(|input| T::parse(context, input)) {
-            return Ok(Either::Second(v));
-        }
-        Length::parse_internal(context, input, AllowedNumericType::NonNegative, AllowQuirks::No).map(Either::First)
+    fn parse<'i, 't>(
+        context: &ParserContext,
+        input: &mut Parser<'i, 't>,
+    ) -> Result<Self, ParseError<'i>> {
+        Ok(NonNegative(Length::parse_internal(
```

Comment:
Nit: Don't we have `parse_non_negative`?

Reply:
Good catch, amended that.